src/ocean/text/xml/PullParser.d

```diff
@@ -539,24 +539,44 @@ version (partialwhite)
 
                 Return the raw value of the current token
 
+                This function may allocate memory. It is always better to
+                use rawValue instead.
+
         ***********************************************************************/
 
         final Ch[] value()
         {
-                return rawValue;
+                version (D_Version2)
+                {
+                    return rawValue.dup;
```

Comment:
Please don't introduce such performance differences between D1 and D2 intentionally - it makes profiling much harder later. If `.dup` is needed, it should be used on both.

However, if you keep in mind comment about `PullParser!(Const!(char))`, it may look like your original assumption mentioned in the commit body is wrong - this method does not inherently return mutable string.